pkg/controller/route/route_controller.go

```diff
@@ -229,25 +254,65 @@ func (rc *RouteController) reconcile(nodes []*v1.Node, routes []*cloudprovider.R
 		}
 	}
 	wg.Wait()
+
+	// after all routes have been created (or not), we start updating
+	// all nodes' statuses with the outcome
+	for _, node := range nodes {
+		wg.Add(1)
+		nodeRoutes := nodeRoutesStatuses[types.NodeName(node.Name)]
+		allRoutesCreated := true
+
+		if len(nodeRoutes) == 0 {
+			go func() {
+				defer wg.Done()
+				rc.updateNetworkingCondition(node, false)
+			}()
+			continue
+		}
+
+		// check if all routes were created. if so, then it should be ready
+		for _, created := range nodeRoutes {
+			if !created {
+				allRoutesCreated = false
+				break
+			}
+		}
+		go func(n *v1.Node) {
```

Comment:
You're closing over the node here, but not in the same loop a few lines up, which is suspicious :-)

Reply:
fair :-)